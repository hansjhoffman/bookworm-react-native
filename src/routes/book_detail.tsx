import React from "react";
import { Image, Platform, StyleSheet, Text, View } from "react-native";
import { NavigationContainerProps } from "react-navigation";
import gql from "graphql-tag";
import { useQuery } from "@apollo/react-hooks";

import { Error, Loading, Stars } from "../components";
import { Colors, Layout } from "../constants";
import { Book } from "../types";
import { Images } from "../../assets";

export const BOOK_QUERY = gql`
  query bookDetail($id: ID!) {
    book(id: $id) {
      id
      authors
      description
      rating
      thumbnail
      title
    }
  }
`;

export function BookDetail({ navigation }: NavigationContainerProps) {
  const id: string = navigation && navigation.getParam("id", "");

  const { data, error, loading } = useQuery(BOOK_QUERY, { variables: { id } });

  if (loading) return <Loading />;
  if (error) return <Error message={error.message} />;

  const book: Book = data.book;

  return (
    <View style={styles.container}>
      <View style={styles.card}>
        <Image
          defaultSource={Images.placeholder}
          resizeMode="contain"
          source={{ uri: book.thumbnail || "WHY????" }}
          style={styles.thumbnail}
        />
        <View style={styles.infoContainer}>
          <Text style={styles.bookTitleText} numberOfLines={2}>
            {book.title}
          </Text>
          <Text style={styles.bookAuthorText} numberOfLines={1}>
            {book.authors}
          </Text>
          <Stars rating={book.rating} />
          <Text style={styles.bookDescriptionText} numberOfLines={4}>
            {book.description}
          </Text>
        </View>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  bookAuthorText: {
    color: Colors.textLight,
    fontFamily: "Lato-Light",
    fontSize: 14,
    marginBottom: Layout.margin_sm,
  },
  bookDescriptionText: {
    color: Colors.textMedium,
    fontFamily: "Lato-Regular",
    fontSize: 16,
    lineHeight: 25,
    marginTop: Layout.margin_sm,
    textAlign: "justify",
  },
  bookTitleText: {
    color: Colors.textDark,
    fontFamily: "Lato-Bold",
    fontSize: 20,
    marginBottom: Layout.margin_sm,
  },
  card: {
    alignItems: "center",
    backgroundColor: Colors.white,
    borderRadius: Layout.cardRadius,
    height: Layout.screen.height * 0.5,
    marginHorizontal: Layout.margin_xl,
    padding: Layout.margin_lg,
    position: "relative",
    ...Platform.select({
      android: {
        elevation: 5,
      },
      ios: {
        shadowColor: Colors.shadow,
        shadowOffset: {
          height: 7,
          width: 0,
        },
        shadowOpacity: 0.1,
        shadowRadius: Layout.cardRadius,
      },
    }),
  },
  container: {
    flex: 1,
    backgroundColor: Colors.white,
    justifyContent: "center",
  },
  icon: {
    height: 25,
    width: 25,
    marginLeft: Layout.margin_lg,
    resizeMode: "contain",
    tintColor: Colors.iconSelected,
  },
  infoContainer: {
    paddingTop: 150,
  },
  thumbnail: {
    borderRadius: 15,
    height: 300,
    overflow: "hidden",
    position: "absolute",
    top: -150,
    width: 200,
  },
});
