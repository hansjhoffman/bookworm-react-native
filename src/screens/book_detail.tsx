import React from "react";
import { Text, View } from "react-native";
import { NavigationContainerProps } from "react-navigation";

export function BookDetail({ navigation }: NavigationContainerProps) {
  const id = navigation && navigation.getParam("id", "");

  return id ? (
    <View>
      <Text>{`Details for book: ${id}`}</Text>
    </View>
  ) : (
    <View>
      <Text>Unable to find book</Text>
    </View>
  );
}
