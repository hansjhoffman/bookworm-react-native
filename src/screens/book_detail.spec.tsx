import React from "react";
import { render } from "@testing-library/react-native";

import { BookDetail } from "./book_detail";

describe("[Screen] book_detail", function() {
  it("should match snapshot", function() {
    const { baseElement } = render(<BookDetail />);

    expect(baseElement).toMatchSnapshot();
  });
});
