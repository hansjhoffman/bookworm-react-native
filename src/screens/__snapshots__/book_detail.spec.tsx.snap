// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Screen] book_detail should match snapshot 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <Text>
        Unable to find book
      </Text>
    </View>
  </View>
</View>
`;
